From 04b142dac0fb7721af877080cd3f5703337c4761 Mon Sep 17 00:00:00 2001
From: Karol Herbst <git@karolherbst.de>
Date: Tue, 26 Feb 2013 20:31:37 +0100
Subject: [PATCH 1/2] add isDedicatedSupportingCoreProfiles function

this function should be used to check if the afns driver is supporting core profiles

v2: remove if statemant by simply returning
---
 libglfork.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/libglfork.cpp b/libglfork.cpp
index ed4d88b..0303275 100644
--- a/libglfork.cpp
+++ b/libglfork.cpp
@@ -327,6 +327,12 @@ static void note_geometry(Display *dpy, Drawable draw, int *width, int *height)
   XGetGeometry(dpy, draw, &root, &x, &y, (unsigned *)width, (unsigned *)height, &bw, &d);
 }
 
+static bool isDedicatedSupportingCoreProfiles(Display *dpy)
+{
+  const char* extensions = primus.afns.glXGetClientString(dpy, GLX_EXTENSIONS);
+  return strstr(extensions, "GLX_ARB_create_context") != NULL;
+}
+
 static void* display_work(void *vd)
 {
   GLXDrawable drawable = (GLXDrawable)vd;
-- 
1.8.2.1

